### PR TITLE
update the catalog preview url to use the catalog service

### PR DESCRIPTION
**Pull request by kellmarie**
ENT-2577

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `make static` has been run to update webpack bundling if any static content was updated
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
- [ ] Version pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)

### CODE REVIEW

File: enterprise/api_client/lms.py [hunk 1]
```diff
@@ -18,7 +18,11 @@
 from django.utils import timezone
 
 from enterprise.constants import COURSE_MODE_SORT_ORDER, EXCLUDED_COURSE_MODES
-from enterprise.utils import NotConnectedToOpenEdX, get_enterprise_worker_user
+try:
+    from enterprise.utils import NotConnectedToOpenEdX, get_enterprise_worker_user
+except ImportError:
+    NotConnectedToOpenEdX = None
+    get_enterprise_worker_user = None
```
Comment (kellmarie) [inline enterprise/api_client/lms.py:25]: So adding a `try` around these import statements fixed the initialization error that I was getting when importing EnterpriseCatalogApiClient in `enterprise.utils.py`

But then running `test_forms.py` has errors with this when initializing the mock catalog info.

Comment (brittneyexline) [reply enterprise/api_client/lms.py:25]: could you remove this change as well as the installed apps added in `enterprise/settings/test.py`?  i would like to see what test failures you got before these changes... i suspect we should be able to resolve by mocking something in a test

Comment (kellmarie) [reply enterprise/api_client/lms.py:25]: sure I'll do that

Comment (kellmarie) [reply enterprise/api_client/lms.py]: Removed those changes

File: enterprise/utils.py [hunk 2]
```diff
@@ -42,10 +42,7 @@
 except ImportError:
     configuration_helpers = None
 
-try:
-    from openedx.core.djangoapps.catalog.models import CatalogIntegration
-except ImportError:
-    CatalogIntegration = None
+from enterprise.api_client.enterprise_catalog import EnterpriseCatalogApiClient
```
Comment (brittneyexline) [inline enterprise/utils.py:45]: so the issue seems to be that importing this class has a cascade effect of importing `JwtLmsApiClient` which is what causes the import error. you will want to copy/paste the try/except ImportError logic that was used for CatalogIntegration above.

File: enterprise/utils.py [hunk 3]
```diff
@@ -1000,29 +997,17 @@ def _get_service_worker(service_worker_username):
...
-                'installed in an Open edX environment.'
-            )
-        )
-    discovery_root_url = CatalogIntegration.current().get_internal_api_url()
-    disc_url = '{discovery_root_url}{search_all_endpoint}?{query_string}'.format(
-        discovery_root_url=discovery_root_url,
-        search_all_endpoint='search/all/',
-        query_string=urlencode(content_filter, doseq=True)
-    )
+    preview_url = EnterpriseCatalogApiClient.API_BASE_URL + EnterpriseCatalogApiClient.GET_CONTENT_METADATA_ENDPOINT.format(uuid)
```
Comment (brittneyexline) [inline enterprise/utils.py:1004]: then here, i would recommend creating a static function in `EnterpriseCatalogApiClient` that returns the preview url so you can mock it in this method and test that function separately. it brings up the question of whether this function is necessary; why not call the function on the client directly?  that's an option too; but at the end of the day the caller will need to make sure `EnterpriseCatalogApiClient` is mocked out in the test, and the function returning the url is tested separately.

Comment (kellmarie) [reply enterprise/utils.py:1004]: Sounds good, I'll do that.

**Pull request merged**